torch/_inductor/codegen/triton.py

```diff
@@ -266,17 +266,55 @@ def triton_reshape(value: str, old_shape: List[str], new_shape: List[str]):
     return f"{value}[{', '.join(expand)}]"
 
 
+# NB: Inheriting from PythonPrinter is somewhat dangerous, because there are a
+# number of operators which Triton "implements", but in a way that is
+# inconsistent with Python semantics (and consistent with C semantics).  We
+# must override all of these, or it is potential silent correctness problem
 class TritonPrinter(PythonPrinter):
-    def _print_floor(self, expr):
+    def _print_TruncToInt(self, expr):
         assert len(expr.args) == 1
         return (
-            f"libdevice.floor({self._print(expr.args[0])}).to({V.kernel.index_dtype})"
+            f"libdevice.trunc({self._print(expr.args[0])}).to({V.kernel.index_dtype})"
         )
 
-    def _print_Trunc(self, expr):
+    def _print_ToFloat(self, expr):
+        assert len(expr.args) == 1
+        return f"{self.paren(self._print(expr.args[0]))}.to(tl.float64)"
+
+    # TODO: This is wrong if one of the inputs is negative.  This is hard to
+    # tickle though, as the inputs are typically positive (and if we can prove
+    # they are positive, we will have used Mod instead, for which this codegen
+    # is right).
```

Comment:
Most of the reason I didn't fix these problems here are (1) they're preexisting problems and (2) I need to write some fairly detailed tests to trigger them

Reply:
1) fair
2) Not really, You can probably just do something like `torch.arange(n, device="cuda") - 1` to get something that's positive and negative but represented by a sympy expression.